### PR TITLE
feat(storage): add support for GCS as storage driver

### PR DESCRIPTION
**Pull request by kishorviswanathan**
### feat(storage): add support for GCS as storage driver

This PR aims to add support for Google Cloud Storage (GCS) as a storage driver.

### CODE REVIEW

**Comment by LouisHaftmann** [comment]
Looks good but we still need to find a way to add tests for this. Are there any gcs emulators for local development?

**Comment by kishorviswanathan** [comment]
@LouisHaftmann I found [this](https://github.com/fsouza/fake-gcs-server), will give it a try.

**Comment by kishorviswanathan** [comment]
@LouisHaftmann I have added the test and that helped me to identify a bug. Thanks for the suggestion.

File: storage-drivers/gcs.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { Storage } from '@google-cloud/storage'
+import { z } from 'zod'
+
+import { defineStorageDriver } from '~/lib/storage/driver'
+
+export const gcsDriver = defineStorageDriver({
+  envSchema: z.object({
+    STORAGE_GCS_BUCKET: z.string().min(1),
+    STORAGE_GCS_SERVICEACCOUNT_KEY: z.string().optional(),
```
Comment (LouisHaftmann) [inline storage-drivers/gcs.ts:9]:
```suggestion
    STORAGE_GCS_SERVICE_ACCOUNT_KEY: z.string().optional(),
```

File: storage-drivers/gcs.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    STORAGE_GCS_ENDPOINT: z.string().optional(),
+  }),
+  async setup(options) {
+    const gcs = new Storage({
+      keyFilename: options.STORAGE_GCS_SERVICEACCOUNT_KEY,
```
Comment (LouisHaftmann) [inline storage-drivers/gcs.ts:14]:
```suggestion
      keyFilename: options.STORAGE_GCS_SERVICE_ACCOUNT_KEY,
```

File: docs/content/2.storage-drivers/gcs.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Don't forget to set the `STORAGE_DRIVER` environment variable to `gcs` to use the GCS storage driver.
+
+#### `STORAGE_GCS_BUCKET`
+
+Example: `gh-actions-cache`
+
+The name of the GCS bucket used for storage.
+
+#### `STORAGE_GCS_SERVICEACCOUNT_KEY`
```
Comment (LouisHaftmann) [inline docs/content/2.storage-drivers/gcs.md:50]:
```suggestion
#### `STORAGE_GCS_SERVICE_ACCOUNT_KEY`
```

File: docs/content/2.storage-drivers/gcs.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ports:
+      - '3000:3000'
+    environment:
+      URL_ACCESS_TOKEN: random_token
+      API_BASE_URL: http://localhost:3000
+
+      STORAGE_DRIVER: gcs
+      STORAGE_GCS_BUCKET: gh-actions-cache
+      # Optional, not required if running in GCP
+      STORAGE_GCS_SERVICEACCOUNT_KEY: /gcp/config/application_default_credentials.json
```
Comment (LouisHaftmann) [inline docs/content/2.storage-drivers/gcs.md:29]:
```suggestion
      STORAGE_GCS_SERVICE_ACCOUNT_KEY: /gcp/config/application_default_credentials.json
```

**Review by LouisHaftmann — CHANGES_REQUESTED** [review]
Just one tiny nitpick: Could we rename `STORAGE_GCS_SERVICEACCOUNT_KEY` to `STORAGE_GCS_SERVICE_ACCOUNT_KEY` for the sake of consistency?

**Comment by kishorviswanathan** [comment]
Done. Renamed it.

**Review by LouisHaftmann — APPROVED** [review]

**Comment by LouisHaftmann** [comment]
Thank you! :heart:

**Comment by LouisHaftmann** [comment]
Will be available in the next release. You can check it out now with the dev image `ghcr.io/falcondev-oss/github-actions-cache-server:dev`

**Pull request merged**